src/main/java/org/ethereum/beacon/discovery/message/handler/NodesHandler.java

```diff
@@ -33,7 +33,7 @@ public void handle(NodesMessage message, NodeSession session) {
         requestInfo.getRemainingNodes() == null
             ? message.getTotal() - 1
             : requestInfo.getRemainingNodes() - 1;
-    if (newNodesCount == 0) {
+    if (newNodesCount <= 0) {
```

Comment:
Why would this be < 0 ?  Shouldn't the responding node always include itself in the response so the total is at least 1?

Reply:
Discv5 only includes nodes *at* the specified distance. So you get the node's own record by itself if you ask for distance 0 and will never get it if you ask for any other distance.  It's weird and mostly quite unhelpful.